dired-rifle.el

```diff
@@ -94,25 +104,30 @@ With `\\[universal-argument] \\[universal-argument]' the output
 is additionally saved to a buffer named *dired-rifle*.
 
 With a numeric prefix argument ARG, run ARGth rifle rule instead
-of the default one (0th).  The output is discarded."
+of the default one (0th).  The output is discarded.
+
+With 0 as numeric argument, switch between focused file and marked files."
   (interactive "P")
   (let ((inhibit-read-only t))
     (let ((output-buffer (when (equal '(16) arg)
                            "*dired-rifle*"))
-          (path (dired-get-filename)))
-      (let ((program-number (if (consp arg)
-                                (string-to-number
-                                 (replace-regexp-in-string
-                                  "^\\([0-9]+\\).*" "\\1"
-                                  (completing-read "Rifle rule: "
-                                                   (rifle-get-rules path)
-                                                   nil
-                                                   t)))
-                              arg)))
-        (message "Launching rifle...")
-        (rifle-open path
-                    program-number
-                    output-buffer)))))
+          (path (if (equal (equal arg 0) dired-rifle-use-marked-files)
+                    (dired-get-filename)
+                  (dired-get-marked-files))))
+      (dolist (p (flatten-tree path))
```

Comment:
I've been thinking… Shouldn't we handle the whole argument list in a single `rifle` call instead of calling `rifle` once for each argument? Is this a conscious decision on your part for some reason?

Reply:
Yes, if you mix MIME-Type in your marking (e.g. .pdf and .txt), the single `rifle` call can open the correct programs. 
I'm working on grouping MIME-Types so that all files of the same type get opened by one instance of a program, but I'm not done yet.

Reply:
That's true, though it's also consistent with the behavior of the original rifle. If we were to fix it, I'd rather fix it there.